src/main/java/com/kalessil/phpStorm/phpInspectionsEA/options/OptionsComponent.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    /**
+     * Creates a Checkbox.
+     *
+     * @param fieldTitle     The title of the field on options panel.
+     * @param defaultValue   The default or current field value stored on inspection.
+     * @param updateConsumer The update consumer (called after item change).
+     */
+    public JCheckBox createCheckbox(final String fieldTitle, final boolean defaultValue, final Consumer<Boolean> updateConsumer) {
+        final JCheckBox createdCheckbox = new JCheckBox(fieldTitle, defaultValue);
+        createdCheckbox.addItemListener((itemEvent) -> updateConsumer.accept(createdCheckbox.isSelected()));
+
+        optionsPanel.add(createdCheckbox, "wrap");
+
+        return createdCheckbox;
+    }
+
+    /**
+     * Creates a new Radio component.
+     *
+     * @param componentDefiner The component definer, that will create the option radios.
+     */
+    public void createRadio(@NotNull final Consumer<RadioComponent> componentDefiner) {
+        final RadioComponent radioComponent = new RadioComponent();
+        componentDefiner.accept(radioComponent);
+    }
+
+    /**
+     * Creates a List.
+     *
+     * @param fieldTitle  The title of this field on panel.
+     * @param listItems   The default or current list items.
+     * @param listUpdater The event called after a modification on list items.
+     */
+    public void createList(final String fieldTitle, @NotNull final List<String> listItems, @NotNull final Runnable listUpdater) {
+        optionsPanel.add(new JLabel(fieldTitle), "wrap");
+        optionsPanel.add((new PrettyListControl(listItems) {
+            protected void fireContentsChanged() {
+                listUpdater.run();
+                super.fireContentsChanged();
+            }
+        }).getComponent(), "pushx, growx");
+    }
+
+    /**
+     * Create the checkboxes used as radios.
+     */
+    public class RadioComponent {
+        /**
+         * Store all Radio Elements handled by this Radio Component.
+         */
+        ArrayList<Element> elements = new ArrayList<>();
+
+        /**
+         * Store the current selected Radio.
+         */
+        @Nullable
+        private Element currentSelection;
+
+        /**
+         * @param fieldTitle     The title of the field on options panel.
+         * @param defaultValue   The default or current field value stored on inspection.
+         * @param updateConsumer The update consumer (called after item change).
+         */
+        public void createOption(final String fieldTitle, final boolean defaultValue, final Consumer<Boolean> updateConsumer) {
+            final Element createElement = new Element(fieldTitle, defaultValue, updateConsumer);
+            elements.add(createElement);
+
+            if (defaultValue) {
+                currentSelection = createElement;
+            }
+        }
+
+        /**
+         * Handle a JCheckBox component and it Update Consumer.
+         */
+        private class Element {
```

Comment:
I don't know what is the best name for this case. Note that it is not really a checkbox, not even a radio button. It stores a checkbox and this consumer. Then the `setSelected()` method can both modify the checkbox checked status and trigger the consumer.

But, I do agree that `Element` is not a good name.
New suggestions?